components/hab/src/command/pkg/uninstall.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs;
+use std::path::Path;
+
+use common::package_graph::PackageGraph;
+use common::ui::{Status, UIWriter, UI};
+use error::{Error, Result};
+use hcore::fs as hfs;
+use hcore::package::{all_packages, PackageIdent, PackageInstall};
+
+//use std::iter::IntoIterator;
+
+///
+/// Delete a package and all dependencies which are not used by other packages
```

Comment:
💭 Should we also be checking wehther this particular package is currently loaded as a service in the supervisor?

Reply:
I was going to add some allow/prohibit functionality next:
* don't delete `hab` :)
* don't delete running services
* don't delete anything specified on CLI or in a ignore file